src/functions/uploadToRSIS/framework/repo/database.ts

```diff
@@ -33,17 +39,24 @@ export const createConnection = async (config: Config): Promise<Connection | und
  * @param bindValues   The bind variables values
  * @throws Various errors that must be caught and logged, including if expected rows proved to be different
  */
-export const execute = async (connection: Connection, sqlQuery: string, expectedRows: number, bindValues?: any):
+export const execute =
+  async (connection: Connection, sqlQuery: string, expectedRows: number, appRef: number, bindValues?: any):
     Promise<void> => {
-  debug(`Executing statement: \n***\n${sqlQuery}\n***`);
+    debug(`Executing statement: \n***\n${sqlQuery}\n***`);
+    let result;
+    try {
+      result = await connection.execute(sqlQuery, bindValues || {}, { autoCommit: true });
 
-  const result = await connection.execute(sqlQuery, bindValues || {}, { autoCommit: true });
+      debug(`${result.rowsAffected} rows updated`);
 
-  debug(`${result.rowsAffected} rows updated`);
+      if (result.rowsAffected !== expectedRows) {
+        const err = `Expected ${expectedRows} rows to be updated, but got ${result.rowsAffected}`;
+      // tslint:disable-next-line:max-line-length
+        error(`Error executing SQL query, expected ${expectedRows} rows to be updated, but got ${result.rowsAffected}, with app ref of ${appRef}`);
+        throw new Error(err);
+      }
+    } catch (err) {
+      error(`Failed to execute sql query with errors: `, err);
+    }
```

Comment:
I think this will give you the same error twice in the logs. We create a string on line 53 then log and error 55, we then throw an error with the string from 53 and log it on 59. 

I would update the error on line 53 and drop the error on 55